toss/fitness/fitness_function_utils.py

```diff
@@ -138,27 +139,19 @@ def compute_space_coverage(number_of_spacecrafts: int, spin_axis: np.ndarray, sp
         j = np.argmin(np.abs(theta[:, np.newaxis] - theta_points), axis=0) # indices along theta axis
         k = np.argmin(np.abs(phi[:, np.newaxis] - phi_points), axis=0) # indices along phi axis
 
-        # Create a new boolean tensor corresponding to the previous visits.
+        # Create a new boolean tensor corresponding to candidate trajectory.
         new_tensor = np.full((len(r), len(theta), len(phi)), False)
-        indices_previous_trajectory  =  np.where(bool_tensor == True)
-        new_tensor[indices_previous_trajectory[0], indices_previous_trajectory[1], indices_previous_trajectory[2]] = True 
-        
-        # Update the new tensor with the proposed candidate trajectory.
         new_tensor[i, j, k] = True
 
-        # Identify number of unique visits in the region of interest
-        indices_unique_visits = np.where(new_tensor == True)
-        n_unique_visits = indices_unique_visits[0].shape[0]
+        # Update the new tensor according to previously spacecraft trajectories to remove overlap in the new candidate trajectory.
+        indices_previous_visits  =  np.where(bool_tensor == True)
+        new_tensor[indices_previous_visits[0], indices_previous_visits[1], indices_previous_visits[2]] = False 
 
-        # Compute ratio of uniquely visited regions to the number of regions defined by the tensor.
-        ratio_unique_visits = n_unique_visits / bool_tensor.size
-
-        # Get weights corresponding to the uniquely visited regions. 
-        r_idx = indices_unique_visits[0] #np.concatenate((previous_visits[0], new_visits[0])) # unique_visits[0]
-        sum_of_weights = np.sum(1/r[r_idx])
+        # Identify number of new visits in the region of interest
+        indices_new_visits = np.where(new_tensor == True)
 
         # Return fitness
-        fitness = ratio_unique_visits + sum_of_weights
+        fitness = np.sum(tensor_weights[indices_new_visits[0]])
```

Comment:
why indices_new_visits[0]? What are the dimensions of it? (also because before you use index 0,1,2 of it)

Reply:
See my comment above